datajoint/expression.py

```diff
@@ -119,17 +121,43 @@ def where_clause(self):
             else " WHERE (%s)" % ")AND(".join(str(s) for s in self.restriction)
         )
 
+    def sorting_clauses(self):
+        if not self._top:
+            return ""
+        limit = self._top.limit
+        order_by = self._top.order_by or ["KEY"]
+        offset = self._top.offset or 0
+
+        if order_by and not (
+            isinstance(order_by, str) or all(isinstance(r, str) for r in order_by)
+        ):
+            raise DataJointError("All order_by attributes must be strings")
+        if offset and limit is None:
+            raise DataJointError("limit is required when offset is set")
+
+        # if 'order_by' passed in a string, make into list
+        if isinstance(order_by, str):
+            order_by = [order_by]
+        # expand "KEY" or "KEY DESC"
+        order_by = list(_flatten_attribute_list(self.primary_key, order_by))
+
+        clause = " ORDER BY " + ", ".join(order_by)
```

Comment:
No need to create a list
```suggestion
        clause = " ORDER BY " + ", ".join(o_flatten_attribute_list(self.primary_key, order_by))
```